### ISSUE TITLE
bug: mssql does not implement `_from_url()`

### ISSUE THREAD

**Issue by grieve54706** [ibis-project/ibis#9415]
### What happened?

The document website has a description of the MSSQL that could use a connection URL to connect.
https://ibis-project.org/backends/mssql#ibis.connect-url-format

But actually backends.mssql inherits `NoUrl` and does not implement `_from_url()`.

### What version of ibis are you using?

9.0.0

### What backend(s) are you using, if any?

MSSQL

### Relevant log output

```sh
self = <ibis.backends.mssql.Backend object at 0x1388b0490>, host = 'localhost'
user = None, password = None, port = 1433, database = None, driver = None
kwargs = {'Trusted_Connection': 'yes'}

    def do_connect(
        self,
        host: str = "localhost",
        user: str | None = None,
        password: str | None = None,
        port: int = 1433,
        database: str | None = None,
        driver: str | None = None,
        **kwargs: Any,
    ) -> None:
        if user is None and password is None:
            kwargs.setdefault("Trusted_Connection", "yes")

>       con = pyodbc.connect(
            user=user,
            server=host,
            port=port,
            password=password,
            database=database,
            driver=driver,
            **kwargs,
        )
E       pyodbc.Error: ('01000', "[01000] [unixODBC][Driver Manager]Can't open lib 'None' : file not found (0) (SQLDriverConnect)")
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by gforsyth** [comment]
Hey @grieve54706 -- thanks for opening this!  This seems like a good starter issue for a new contributor, so we might leave it open for a little while, but I'm sure someone will pick it up eventually.

Let us know if it becomes a blocker for you!

**Comment by grieve54706** [comment]
Hi @gforsyth,

I wish I had the time to help implement this feature. Would you mind if I submit a PR to fix it when I have some free time?

**Comment by gforsyth** [comment]
Hey @grieve54706!  Definitely!  It's possible someone else might pick it up, but I'm happy to wait on it.

**Comment by pangeran-bottor** [comment]
hi @grieve54706, have you already started working on this? if not and if it's ok, I would like to take on this

**Comment by grieve54706** [comment]
Hi @pangeran-bottor, I'm not ready. You can take this. Thank you for your time and help.

**Comment by cpcloud** [comment]
Closed by #9894.
